server/data/mappers/actionPlanMapper.ts

```diff
@@ -1,19 +1,32 @@
-import type { ActionPlanResponse, GoalResponse, StepResponse } from 'educationAndWorkPlanApiClient'
+import type { ActionPlanResponse, GetGoalsResponse, GoalResponse, StepResponse } from 'educationAndWorkPlanApiClient'
 import type { ActionPlan, Goal, Step } from 'viewModels'
 import dateComparator from '../../routes/dateComparator'
 
 const toActionPlan = (actionPlanResponse: ActionPlanResponse, problemRetrievingData: boolean): ActionPlan => {
-  const goalReferencesInCreationDateOrder = goalReferencesSortedByCreationDate(actionPlanResponse.goals)
+  const goals = toOrderedGoals(actionPlanResponse.goals)
   return {
     prisonNumber: actionPlanResponse.prisonNumber,
-    goals: actionPlanResponse.goals.map((goal: GoalResponse) => {
-      const goalSequenceNumber = goalReferencesInCreationDateOrder.indexOf(goal.goalReference) + 1
-      return toGoal(goal, goalSequenceNumber)
-    }),
+    goals,
     problemRetrievingData,
   }
 }
 
+const toGoals = (response: GetGoalsResponse): Goal[] => {
+  return toOrderedGoals(response.goals)
+}
+
+/**
+ * Sets a goal sequence by creation date (oldest = 1) and returns them in creation order (newest first)
+ */
+function toOrderedGoals(goals: GoalResponse[]): Goal[] {
+  return [...goals]
+    .sort((left: GoalResponse, right: GoalResponse) =>
+      dateComparator(toDate(left.createdAt), toDate(right.createdAt), 'ASC'),
+    )
+    .map((goal, index) => toGoal(goal, index + 1))
+    .reverse()
```

Comment:
It was this call to `.reverse()` that made me look at the diff (over and over again! 🤣 ) to see how my original code was flawed! 🤣.
FWIW, instead of using `.reverse`, you could pass `DESC` into the dateComparator as an arg instead
I guess there's a very minor negligible performance benefit by not calling `reverse`, but I doubt we'd ever see and tangible benefit from it.
And if it wasn't for the call to `reverse` I might not have looked so hard at the code to see where my original code was shonky! 🤣

Reply:
It's actually even more confusing because it numbers them in ASC date order but displays them in DESC date order, hence the reverse. The numbering is going with the new overview page so hopefully we can delete most of this mess then anyway.

Reply:
Yes, good point about the numbering; and yes, once we can start on the new design and remove the numbering this will be so much easier 👍